### PR TITLE
LINK-2024 | improve test coverage

### PR DESCRIPTION
**Pull request by jorilindell**
## Description
- Adds tests for the components and the functions which were not fully tested.

## Closes
[LINK-2024](https://helsinkisolutionoffice.atlassian.net/browse/LINK-2024)

[LINK-2024]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-2024?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**LINKEDREGISTRATIONS-UI branch is deployed to platta: https://linkedregistrations-pr167-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedregistrations-pr167-ui.dev.hel.ninja :laughing::tada::tada::tada:

File: src/domain/signupGroup/validation.ts [hunk 1]
```diff
@@ -290,23 +291,22 @@ export const showErrors = ({
   }
 };
 
-const getFocusableFieldId = (
+type FieldList = Array<string | RegExp>;
+
+const isInFieldList = (fieldList: FieldList, fieldName: string): boolean =>
```
Comment (harriris-vincit) [inline src/domain/signupGroup/validation.ts:296]: This together with adding the new constant `SIGNUP_GROUP_FORM_CHECKBOX_GROUP_FIELDS` to `src/domain/signupGroup/constants.ts` could be in their own commit.

**Comment by terovirtanen** [comment]
**LINKEDREGISTRATIONS-UI branch is deployed to platta: https://linkedregistrations-pr167-ui.dev.hel.ninja** :rocket::rocket::rocket:

File: src/domain/signup/__tests__/query.test.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+  await waitFor(() => expect(result.current.data).toEqual(signup));
+});
+
+test('should return error for the failing signup query', async () => {
+  console.error = jest.fn();
+  const error = { errorMessage: 'Failed to fetch signup' };
+  setQueryMocks(
+    rest.get(`*/signup/${TEST_SIGNUP_ID}`, (req, res, ctx) =>
+      res(ctx.status(404), ctx.json(error))
```
Comment (harriris-vincit) [inline src/domain/signup/__tests__/query.test.ts:76]: Non-mandatory suggestion: testing for other status code than `404` wouldn't improve coverage, but could catch some future problems if the error handling of fetching a signup is changed. So maybe testing for some possible other statuses like `401`, `403`, `500` and `502` could be added.

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedregistrations-pr167-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**LINKEDREGISTRATIONS-UI branch is deployed to platta: https://linkedregistrations-pr167-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedregistrations-pr167-ui.dev.hel.ninja :laughing::tada::tada::tada:

File: src/domain/signup/__tests__/query.test.ts (hunk 2 above)
Comment (jorilindell) [reply src/domain/signup/__tests__/query.test.ts]: Added also tests to other status codes
